stan/math/prim/meta/is_autodiff.hpp

```diff
@@ -15,88 +22,119 @@ namespace stan {
  * @ingroup type_trait
  */
 template <typename T>
-struct is_autodiff
+struct is_autodiff_scalar
```

Comment:
I like the name change. Just curious, is there any other reason the name changed other than clarity? (I don't know if there were conflicts or compiler errors that you had to work through.)